src/controllers/import.js

```diff
@@ -156,6 +190,25 @@ function ImportController(context) {
         urls, options, importScript, customHeaders,
       } = multipartFormData;
 
+      const scopes = profile.getScopes();
+
+      // We only check if the URLs belong to the allowed domains if the user has the write scope
+      // We do not need to check the domains for users with scope: all_domains
+      if (!scopes.some((scope) => scope.name === SCOPE.ALL_DOMAINS)) {
```

Comment:
You could use [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap).

```
if (!scopes.some(scope => scope.name === SCOPE.ALL_DOMAINS)) {
  const allowedDomains = scopes
    .filter(scope => scope.name === SCOPE.WRITE && scope.domains && scope.domains.length > 0)
    .flatMap(scope => scope.domains.map(getDomain));

  if (allowedDomains.length === 0) {
    throw new ErrorWithStatusCode('Missing domain information', STATUS_UNAUTHORIZED);
  }

  isUrlInBaseDomains(urls, allowedDomains);
}
```

Reply:
Nice! I'll add that in!

Reply:
A little helper function to perform this check might improve readability, such as:
```suggestion
      if (hasAllDomainsScope(scopes)) {
```